fadecandy_driver/src/fadecandy_driver.cpp

```diff
@@ -0,0 +1,300 @@
+  int rc = 0;
+  unsigned count = 0;
+
+  rc = libusb_init(&context_);
+
+  count = libusb_get_device_list(context_, &list);
+  for (size_t idx = 0; idx < count; ++idx)
+  {
+    libusb_device* device = list[idx];
+    libusb_device_descriptor desc = { 0 };
+
+    rc = libusb_get_device_descriptor(device, &desc);
+    assert(rc == 0);
+    if (desc.idVendor == USB_VENDOR_ID && desc.idProduct == USB_PRODUCT_ID)
+    {
+      fadecandy_device_ = device;
+      fadecandy_device_descriptor_ = desc;
+    }
+  }
+}
+
+std::vector<std::vector<unsigned char>>
+FadecandyDriver::makeVideoUsbPackets(std::vector<std::vector<Color>> led_array_colors)
+{
+  int led_index = 0;
+  std::vector<Color> all_led_colors(LEDS_PER_STRIP * NUM_STRIPS, { 0, 0, 0 });
+  for (size_t i = 0; i < led_array_colors.size(); i++)
+  {
+    for (size_t j = 0; j < led_array_colors[i].size(); j++)
+    {
+      led_index = (i * LEDS_PER_STRIP) + j;
+      all_led_colors[led_index] = led_array_colors[i][j];
+    }
+  }
+  assert(all_led_colors.size() == LEDS_PER_STRIP * NUM_STRIPS);
+  std::vector<std::vector<unsigned char>> packets;
+  std::vector<unsigned char> packet;
+  std::vector<Color> packet_leds;
+  std::vector<Color> remaining_leds = all_led_colors;
+  std::vector<int> color_bytes;
+  int control;
+
+  while (remaining_leds.size() > 0)
+  {
+    color_bytes.clear();
+    packet.clear();
+
+    if (remaining_leds.size() < LEDS_PER_PACKET)
+    {
+      packet_leds.assign(remaining_leds.begin(), remaining_leds.end());
+      remaining_leds.erase(remaining_leds.begin(), remaining_leds.end());
+    }
+    else
+    {
+      packet_leds.assign(remaining_leds.begin(), remaining_leds.begin() + LEDS_PER_PACKET);
+      remaining_leds.assign(remaining_leds.begin() + LEDS_PER_PACKET, remaining_leds.end());
+    }
+
+    control = packets.size() | PACKET_TYPE_VIDEO;
+    if (remaining_leds.size() == 0)
+    {
+      control |= FINAL_PACKET_BIT;
+    }
+
+    for (size_t i = 0; i < packet_leds.size(); i++)
+    {
+      color_bytes.push_back(packet_leds[i].r);
+      color_bytes.push_back(packet_leds[i].g);
+      color_bytes.push_back(packet_leds[i].b);
+    }
+
+    if (63 - color_bytes.size() > 0)
+    {
+      int j = 63 - color_bytes.size();
+      for (int i = 0; i < j; i++)
+      {
+        color_bytes.push_back(0);
+      }
+    }
+
+    for (size_t i = 0; i < color_bytes.size(); i++)
+    {
+      int byts_per_int = 1;
+      unsigned char buffer[byts_per_int];
+      std::vector<unsigned char> bufferv;
+      bufferv = intToCharArray(buffer, color_bytes[i], byts_per_int);
+      packet.insert(packet.end(), bufferv.begin(), bufferv.end());
+    }
+
+    // add control byte
+    packet.insert(packet.begin(), static_cast<unsigned char>(control));
+
+    assert(packet.size() == USB_PACKET_SIZE);
+    packets.push_back(packet);
+  }
+  return packets;
+}
+
+std::vector<std::vector<unsigned char>> FadecandyDriver::makeLookupTablePackets(std::vector<int> red_lookup_values,
```

Comment:
For this function the comments are similar as above. Also please check if your loop logic is efficient.